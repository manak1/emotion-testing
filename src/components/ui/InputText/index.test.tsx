import userEvent from "@testing-library/user-event"
import { render, screen } from "~/test/test-utils"

import * as stories from "./index.stories"
import { composeStories } from "@storybook/testing-react"

const { Default, Optional } = composeStories(stories)

describe("InputTextのテスト", () => {
  const onChange = jest.fn()
  it("snapshot", () => {
    const { asFragment } = render(<Default onChange={onChange} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("labelが表示される", () => {
    render(<Default onChange={onChange} />)
    expect(screen.getByLabelText("ラベル")).toBeInTheDocument()
  })
  it("初期値が表示される", () => {
    render(<Default onChange={onChange} />)
    expect(screen.getByRole("textbox")).toHaveValue("test")
  })
  it("idが設定されている", () => {
    render(<Default onChange={onChange} />)
    expect(screen.getByRole("textbox")).toHaveAttribute("id", "test")
  })
  it("任意の文字が表示される", () => {
    render(<Optional onChange={onChange} />)
    expect(screen.getByText("任意")).toBeInTheDocument()
  })
  it("入力時にonChangeが呼ばれる", async () => {
    render(<Default onChange={onChange} />)
    const input = screen.getByRole("textbox")
    userEvent.type(input, "テスト")
    expect(onChange).toBeCalledTimes(3)
  })
})
