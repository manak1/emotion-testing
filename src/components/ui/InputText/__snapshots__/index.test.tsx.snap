// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextのテスト snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-j7qwjs"
  >
    <div
      class="css-69i1ev"
    >
      <label
        class="css-1lu35ut"
        for="test"
      >
        ラベル
      </label>
    </div>
    <input
      autocomplete="off"
      class="css-glhwro"
      id="test"
      type="text"
      value="test"
    />
  </div>
</DocumentFragment>
`;
